parser/parse.go

```diff
@@ -304,18 +307,25 @@ func (b *Builder) addDir(dir string, userRequested bool) error {
 		b.absPaths[pkgPath] = buildPkg.Dir
 	}
 
-	for _, n := range buildPkg.GoFiles {
-		if !strings.HasSuffix(n, ".go") {
-			continue
-		}
-		absPath := filepath.Join(buildPkg.Dir, n)
-		data, err := ioutil.ReadFile(absPath)
-		if err != nil {
-			return fmt.Errorf("while loading %q: %v", absPath, err)
-		}
-		err = b.addFile(pkgPath, absPath, data, userRequested)
-		if err != nil {
-			return fmt.Errorf("while parsing %q: %v", absPath, err)
+	directories := [][]string{buildPkg.GoFiles}
+	if b.IncludeTestFiles {
+		directories = append(directories, buildPkg.TestGoFiles)
```

Comment:
why so complicated? just use a `files := []string{}` and append `GoFiles...` and `TestGoFiles...`.